lib/MahoCLI/Commands/SysEncryptionKeyRegenerate.php

```diff
@@ -0,0 +1,233 @@
+        $encryptedData = $readConnection->fetchAll($select);
+
+        if (empty($encryptedData)) {
+            $output->writeln('<info>No encrypted configurations to re-encrypt.</info>');
+            return Command::SUCCESS;
+        }
+
+        $outputTable = new Table($output);
+        $outputTable->setHeaders(['config_id', 'scope', 'scope_id', 'path']);
+        foreach ($encryptedData as $encryptedDataRow) {
+            $writeConnection->update(
+                $table,
+                ['value' => $this->encrypt($this->decrypt($encryptedDataRow['value']))],
+                ['config_id = ?' => $encryptedDataRow['config_id']],
+            );
+
+            $outputTable->addRow([
+                $encryptedDataRow['config_id'],
+                $encryptedDataRow['scope'],
+                $encryptedDataRow['scope_id'],
+                $encryptedDataRow['path'],
+            ]);
+        }
+
+        $output->writeln('<info>The following configurations were just re-encrypted, make sure to test all of them.</info>');
+        $outputTable->render();
+
+        Mage::dispatchEvent('encryption_key_regenerated', [
+            'output' => $output,
+            'encrypt_callback' => [$this, 'encrypt'],
+            'decrypt_callback' => [$this, 'decrypt'],
+        ]);
+
+        if (\Composer\InstalledVersions::isInstalled('phpseclib/mcrypt_compat')) {
+            $output->writeln('');
+            $output->writeln('<error>Warning: phpseclib/mcrypt_compat is installed. This package can cause encryption issues and should be removed.</error>');
+            $output->writeln('<error>Please remove it using: composer remove phpseclib/mcrypt_compat</error>');
+            $output->writeln('');
+        }
+
+        return Command::SUCCESS;
+    }
+
+    private function decrypt(#[\SensitiveParameter] string $data): string
+    {
+        if ($this->isOldEncryptionKeyM1 && function_exists('mcrypt_module_open')) {
+            $key = $this->oldEncryptionKey;
+            $handler = mcrypt_module_open(MCRYPT_BLOWFISH, '', MCRYPT_MODE_ECB, ''); // @phpstan-ignore constant.notFound,constant.notFound
+            $initVector = mcrypt_create_iv(mcrypt_enc_get_iv_size($handler), MCRYPT_RAND); // @phpstan-ignore function.notFound,function.notFound,constant.notFound
+            mcrypt_generic_init($handler, $key, $initVector); // @phpstan-ignore function.notFound
+            $data = mdecrypt_generic($handler, (string) base64_decode((string) $data)); // @phpstan-ignore function.notFound
+            mcrypt_generic_deinit($handler); // @phpstan-ignore function.notFound
+            mcrypt_module_close($handler); // @phpstan-ignore function.notFound
+            return str_replace("\x0", '', trim($data));
+        }
+
+        try {
+            $decoded = sodium_base642bin($data, SODIUM_BASE64_VARIANT_ORIGINAL);
+        } catch (SodiumException $e) {
+            $exception = new Exception('Invalid base64 encoding: ' . $e->getMessage());
```

Comment:
```suggestion
        } catch (\SodiumException $e) {
            $exception = new \Exception('Invalid base64 encoding: ' . $e->getMessage());
```

This should fix the PHPStan error.